src/cpp/src/llm_pipeline.cpp

```diff
@@ -339,11 +400,354 @@ std::pair<std::string, Any> generation_config(const GenerationConfig& config) {
+    ov::InferRequest m_prefill_request;
+};
+
+NPULLMPipelineImpl::NPULLMPipelineImpl(
+    const std::filesystem::path& path,
+    const ov::genai::Tokenizer& tokenizer,
+    const ov::AnyMap& config
+) : LLMPipelineImplBase(tokenizer,
+                        from_config_json_if_exists(path),
+                        chat_template_from_tokenizer_json_if_exists(path)) {
+    /* NB: NPU-friendly LLM pipeline consists of two models,
+       first to process the input prompt (prefill), second to use in generation loop (kvcache)
+
+       Initialization assumes multiple steps:
+       1) Read the template model and clone it for the further use as kvcache and prefill models
+       2) Expose KV-cache input and output layers for the kvcache model
+       3) Reshape both models to static shape
+       4) Add slices to KV-cache inputs for kvcache model, this will make input and output KV-cache
+          layers to have the same shape and allow outputs writes directly to inputs for the next iteration.
+       5) Compile both models for NPUW
+       6) Initialize input tensors for kvcache and prefill models
+    */
+
+    ov::Core core;
+    // (1) Read the template model and clone it
+    auto kvcache_model = core.read_model(path / "openvino_model.xml");
+    auto prefill_model = kvcache_model->clone();
+    // TODO: (2) Expose KV-cache input and output layers
+    // ov::pass::ExposeKVCacheFromModel().run_on_model(kvcache_model);
+    prefill_model->set_friendly_name(kvcache_model->get_friendly_name() + "_prefill");
+    // (3) Reshape both models to static shape
+    // FIXME: There must be better logic than just hardcoded values
+    m_kvcache_desc = KVCacheDesc { 1024u, 0u };
+    const uint32_t max_prompt_size = m_kvcache_desc.total_size;
+    const uint32_t max_kvcache_size = m_kvcache_desc.total_size;
+    reshape_to_static(prefill_model, max_prompt_size, max_kvcache_size);
+    reshape_to_static(kvcache_model, 1u, max_kvcache_size);
+    // (4) Add slices to kvcache model
+    kvcache_model = add_slices_to_kvcache_inputs(kvcache_model);
+    // (5) Compiled both models for NPUW
+    std::map<std::string, std::string> cfg = { {"NPU_USE_NPUW", "YES"},
+                                               {"NPU_COMPILATION_MODE_PARAMS", "compute-layers-with-higher-precision=Sqrt,Power,ReduceMean,Add" },
+                                               {"NPUW_FOLD", "YES"},
+                                               {"NPUW_DCOFF_TYPE", "f16"},
+                                               {"NPUW_DCOFF_SCALE", "YES"},
+                                               {"NPUW_ONLINE_PIPELINE", "NONE"} };
+    ov::AnyMap properties{cfg.begin(), cfg.end()};
+    m_prefill_request = core.compile_model(prefill_model, "NPU", properties).create_infer_request();
+    m_kvcache_request = core.compile_model(kvcache_model, "NPU", properties).create_infer_request();
+    // (6) Initialize tensors
+    prepareForNewConversation();
```

Comment:
```suggestion
    prepare_for_new_conversation();
```
can we use snake case?

Reply:
yea, you'are right, will change